### PR TITLE
Save simulations locally

### PR DESCRIPTION
**Pull request by magopian**
Cf https://www.notion.so/ETQ-toutes-les-marques-JV-pouvoir-enregistrer-des-sc-narios-d5266a1037c44383b79fd7c8e7af4533

### CODE REVIEW

**Comment by magopian** [comment]
Voilà à quoi ça ressemble pour le moment

Capture d’écran 2022-03-21 à 17 22 58

**Comment by magopian** [comment]
Voici à quoi ça ressemble avec les retours de Nico 
Capture d’écran 2022-03-22 à 16 37 50

File: src/Views/Format.elm [hunk 1]
```diff
@@ -118,6 +126,11 @@ percent =
     formatRichFloat 2 "%"
 
 
+percentAsString : Float -> String
+percentAsString value =
+    String.fromInt (round (value * 100)) ++ "\u{202F}%\u{00A0}"
```
Comment (n1k0) [inline src/Views/Format.elm:131]: I'd rather see `ratioToPercentString`, converting a `Unit.Ratio`, expressed as a percentage string

File: src/Data/Inputs.elm [hunk 2]
```diff
@@ -165,6 +168,107 @@ toQuery inputs =
...
+    materials
+        |> List.filter (\{ share } -> Unit.ratioToFloat share > 0)
+        |> List.map
+            (\{ material, share, recycledRatio } ->
+                Format.formatFloat 0 (Unit.ratioToFloat share * 100)
+                    ++ "% "
+                    ++ (material
+                            |> Material.fullName
+                                (material.recycledProcess |> Maybe.map (always recycledRatio))
+                       )
```
[the remark this answers is not in the document]
Comment (n1k0) [reply src/Data/Inputs.elm:209]: Ohhh no I see that we're checking for `recycledProcess` only to check if the material is recyclable… I see the point, but it's rather confusing to infer, so I'd suggest we at least simplify the API here.

[the remark this answers is not in the document]
Comment (n1k0) [reply src/Data/Inputs.elm:209]: I mean by updating `Material.fullName` which should take care of ensuring that a meterial is recyclable before proceeding with any passed recycled ratio.

File: src/Data/Inputs.elm [hunk 3]
```diff
@@ -165,6 +168,107 @@ toQuery inputs =
...
+                ratio
+                    |> Unit.ratioToFloat
+                    |> Format.percentAsString
+                    |> (\percent ->
+                            "\u{202F}%\u{00A0} (avec un procédé " ++ percent ++ " majorant)"
+                       )
+
+
+airTransportRatioAsString : Maybe Unit.Ratio -> String
+airTransportRatioAsString maybeRatio =
```
[the remark this answers is not in the document]
Comment (magopian) [reply src/Data/Inputs.elm:235]: What should we do in the case there's no airTransportRatio? In the `Step.airTransportRatioToString` I guess it should still return "aucun transport aérien", but not in `Inputs.airTransportRatioToString` (as reviewed in https://github.com/MTES-MCT/wikicarbone/pull/105#discussion_r833008809)?

[the remark this answers is not in the document]
Comment (n1k0) [reply src/Data/Inputs.elm:235]: These are for two distinct use cases:

- as a rangeslider label (where we *must* say when no transport ratio is applied)
- as a chart description (where we shouldn't)

So either we write two different functions, or we add a parameter allowing to specify if we want exhaustiveness or not.

[the remark this answers is not in the document]
Comment (magopian) [reply src/Data/Inputs.elm]: I'm going to try something in a different commit, let's see if it's better

[the remark this answers is not in the document]
Comment (magopian) [reply src/Data/Inputs.elm]: What do you think? https://github.com/MTES-MCT/wikicarbone/pull/105/commits/db800acec5eed7d8f6cc9204ffd4082f7ae4ef91

[the remark this answers is not in the document]
Comment (magopian) [reply src/Data/Inputs.elm]: Ok, reverted after talking with Nico ;)

**Review by n1k0 — APPROVED** [review]
Looks good, thanks ! 👍

**Pull request merged**